apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.SingleChannel.cs

```diff
@@ -276,10 +276,19 @@ private void StopStreamingPull()
                 // Ignore all errors; the stream may be in any state.
                 try
                 {
-                    _registerTaskFn(_pull.WriteCompleteAsync());
+                    _registerTaskFn(_pull.WriteCompleteAsync().ContinueWith(_ =>
+                        {
+                            try
+                            {
+                                _pull.Dispose();
+                            }
+                            finally
+                            {
+                                _pull = null;
+                            }
+                        }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, _taskHelper.TaskScheduler));
```

Comment:
I'm not sure why we'd need the dispose to Execute on the same thread as the WriteCompleteAsync?

Reply:
I believe that `SingleChannel` is effectively single-threaded.  This code keeps it consistent.

Reply:
Then in that case I would prefer this code relying on whatever existing code is there that enforces the single threadness. That seems to be `Add(Task task, NextAction next)` that takes the first and second action. If you follow the code, you'll see that `next` will execute even if `task` failed.